### DISCUSSION TITLE
METIS partitioning deterministic or probabilistic?

### DISCUSSION THREAD

**Discussion by merelkuijs** [pyg-team/pytorch_geometric#8167] [Q&A] [answered]
The `ClusterData` class uses METIS partitioning to assign nodes to partitions. The actual partitioning happens in `torch.ops.torch_sparse.partition`, but I am having a hard time finding the source code. 

Does anyone know if repeated calls lead to different partitions, i.e if the partitioning is probabilistic?

**Comment by rusty1s** [comment, marked as answer]
You can find the source code here: https://github.com/rusty1s/pytorch_sparse/blob/master/csrc/cpu/metis_cpu.cpp

The partitioning should be deterministic.
